packages/smart-contracts/test/contracts/BatchConversionPayments.test.ts

```diff
@@ -0,0 +1,840 @@
+        .add(BigNumber.from(fee).mul(conversionRate).mul(nPayment))
+        .mul(BATCH_CONV_FEE)
+        .div(BATCH_DENOMINATOR)
+        // fee within the invoice: .1% of the amount,
+        .add(BigNumber.from(fee).mul(conversionRate).mul(nPayment));
+    fee;
+    const expectedFromDAIBalanceDiff = expectedToDAIBalanceDiff
+      .add(expectedFeeDAIBalanceDiff)
+      .mul(-1);
+    return [
+      expectedFromDAIBalanceDiff.div('1000000'), // divide by 1 million because we used millionDai
+      expectedToDAIBalanceDiff.div('1000000'),
+      expectedFeeDAIBalanceDiff.div('1000000'),
+    ];
+  };
+
+  /** No conversion */
+  const getExpectedERC20Balances = (amount: number, fee: number, nPayment: number) => {
+    const expectedToDAIBalanceDiff = BigNumber.from(amount).mul(nPayment);
+    const expectedFeeDAIBalanceDiff =
+      // fee added by the batch
+      expectedToDAIBalanceDiff
+        .mul(BATCH_FEE)
+        .div(BATCH_DENOMINATOR)
+        // fee within the invoice: .1% of the amount,
+        .add(BigNumber.from(fee).mul(nPayment));
+    fee;
+    const expectedFromDAIBalanceDiff = expectedToDAIBalanceDiff
+      .add(expectedFeeDAIBalanceDiff)
+      .mul(-1);
+    return [expectedFromDAIBalanceDiff, expectedToDAIBalanceDiff, expectedFeeDAIBalanceDiff];
+  };
+
+  /** Compares the expected delta-balances with the one it computes for from, to and fee addresses. */
+  const expectERC20BalanceDiffs = async (
+    token: 'DAI' | 'FAU',
+    initialFromBalance: BigNumber,
+    initialToBalance: BigNumber,
+    initialFeeBalance: BigNumber,
+    expectedFromBalanceDiff: BigNumber,
+    expectedToBalanceDiff: BigNumber,
+    expectedFeeBalanceDiff: BigNumber,
+  ) => {
+    const testERC20 = token === 'FAU' ? fauERC20 : daiERC20;
+    // Get balances
+    const [fromBalance, toBalance, feeBalance, batchBalance] = await getERC20Balances(testERC20);
+    // Compare balance changes to expected values
+    const fromBalanceDiff = BigNumber.from(fromBalance).sub(initialFromBalance);
+    const toBalanceDiff = BigNumber.from(toBalance).sub(initialToBalance);
+    const feeBalanceDiff = BigNumber.from(feeBalance).sub(initialFeeBalance);
+
+    expect(toBalanceDiff).to.equals(expectedToBalanceDiff, `toBalanceDiff in ${token}`);
+    expect(feeBalanceDiff).to.equals(expectedFeeBalanceDiff, `feeBalanceDiff in ${token}`);
+    expect(fromBalanceDiff).to.equals(expectedFromBalanceDiff, `fromBalanceDiff in ${token}`);
+    expect(batchBalance).to.equals('0', `batchBalance in ${token}`);
+  };
+
+  /** Compares the expected delta-balances with the one it computes for from, to and fee addresses. */
+  const expectETHBalanceDiffs = async (
+    ethAmount: BigNumber,
+    ethFeeAmount: BigNumber,
+    feeApplied = BATCH_CONV_FEE,
+    initialFromETHBalance: BigNumber,
+    initialToETHBalance: BigNumber,
+    initialFeeETHBalance: BigNumber,
+  ) => {
+    const receipt = await tx.wait();
+    const gasUsed = receipt.gasUsed.mul(2 * 10 ** 10);
```

Comment:
Why `2 * 10 ** 10` ?

Reply:
10^10: When forking a remote network, if the remote network uses EIP-1559, the first local block will use the right baseFeePerGas according to the EIP, otherwise "**10_000_000_000**" is used. from: https://hardhat.org/hardhat-network/docs/reference

Reply:
it is the gasPrice, it is updated